diesel/Cargo.toml

```diff
@@ -31,6 +31,7 @@ num-traits = { version = "0.1.35", optional = true }
 num-integer = { version = "0.1.32", optional = true }
 bigdecimal = { version = "0.0.10", optional = true }
 bitflags = { version = "1.0", optional = true }
+r2d2 = { version = ">= 0.7, < 0.9", optional = true }
```

Comment:
There is a problem here, the r2d2 version you chose here [doesn't make `r2d2::Error` public](https://docs.rs/r2d2/0.7.4/r2d2/) and make the diesel r2d2 reexport to not compile. In the diesel 0.8 version, [the `Error` type is now public](https://docs.rs/r2d2/0.8.2/r2d2/).

```rust
error[E0603]: trait `Error` is private
  --> /Users/crenault/.cargo/registry/src/github.com-1ecc6299db9ec823/diesel-1.1.1/src/r2d2.rs:11:22
   |
11 | pub type PoolError = self::r2d2::Error;
   |                      ^^^^^^^^^^^^^^^^^
```

Reply:
Whooops, it seems to be fixed in the master...

Reply:
Yes, [this PR](https://github.com/diesel-rs/diesel/pull/1513) fixed that problem.

Reply:
It is possible to fix this locally by executing `cargo update -p r2d2`